app/utils/context_runner.rb

```diff
@@ -131,6 +127,8 @@ def next_mw(ctx)
       [new_ctx, mw, :error]
     elsif !new_ctx[:enter_queue].empty?
       mw = new_ctx[:enter_queue].pop
+      new_ctx[:leave_stack].unshift(mw)
```

Comment:
This makes a lot of sense. I also confirmed from this is how Pedestal [handles it](https://github.com/pedestal/pedestal/blob/master/interceptor/src/io/pedestal/interceptor/chain.clj#L151).